packages/browser/src/tracing/browserTracingIntegration.ts

```diff
@@ -459,16 +469,18 @@ export const browserTracingIntegration = ((_options: Partial<BrowserTracingOptio
               return;
             }
 
-            if (from !== to) {
-              startingUrl = undefined;
+            startingUrl = undefined;
+
+            // We wait a tick here to ensure that WINDOW.location.pathname is updated
+            setTimeout(() => {
```

Comment:
m: I'm a bit worried about delaying the span start for a tick. I guess this comes down to the router or implementation but what if users synchronously push a new state and e.g. make a `fetch` call directly afterwards? Would we still catch the span?
i think alternatively, we could still start the span directly but call `updateSpanName` once more in the `setTimeout` closure.
Not ideal in terms of DSC consistency but I'd rather have the span than miss it. WDYT?